source/s3_auto_ranged_get.c

```diff
@@ -739,17 +791,32 @@ static void s_s3_auto_ranged_get_request_finished(
 
             copy_http_headers(request->send_data.response_headers, response_headers);
 
-            /* If this request is a part, then the content range isn't applicable. */
             if (request->request_tag == AWS_S3_AUTO_RANGE_GET_REQUEST_TYPE_GET_OBJECT_WITH_RANGE ||
                 request->request_tag == AWS_S3_AUTO_RANGE_GET_REQUEST_TYPE_GET_OBJECT_WITH_PART_NUMBER_1) {
-                /* For now, we can assume that discovery of size via the first part of the object does not apply to
-                 * breaking up a ranged request. If it ever does, then we will need to repopulate this header. */
-                AWS_ASSERT(!auto_ranged_get->initial_message_has_range_header);
-                aws_http_headers_erase(response_headers, g_content_range_header_name);
+
+                if (!auto_ranged_get->initial_message_has_range_header) {
+                    /* content range isn't applicable. */
+                    aws_http_headers_erase(response_headers, g_content_range_header_name);
+                } else {
```

Comment:
utterly trivial: in my opinion, code reads nicer as `if(X)/else`, vs `if (!X)/else`

Reply:
Thanks, updated.